File: KargoPaylasimJava/app/src/main/java/com/example/kargopaylasimjava/util/PhoneUtil.java
package com.example.kargopaylasimjava.util;

public final class PhoneUtil {
    private PhoneUtil() {}

    public static String normalizeTrToE164(String input) {
        if (input == null) return "";
        String raw = input.trim().replace(" ", "").replace("-", "");
        if (raw.isEmpty()) return raw;

        if (raw.startsWith("+90")) return "+90" + raw.substring(3).replaceAll("\\D+", "");
        if (raw.startsWith("90") && raw.length() >= 12) return "+90" + raw.substring(2).replaceAll("\\D+", "");
        if (raw.startsWith("0")) return "+90" + raw.substring(1).replaceAll("\\D+", "");
        if (raw.startsWith("5")) return "+90" + raw.replaceAll("\\D+", "");
        return raw;
    }

    public static boolean isLikelyTrPhoneE164(String phone) {
        if (phone == null || !phone.startsWith("+90")) return false;
        String digits = phone.substring(1).replaceAll("\\D+", ""); // "90xxxxxxxxxx"
        return digits.length() == 12;
    }
}

File: KargoPaylasimKotlin/app/src/main/java/com/example/kargopaylasimkotlin/util/PhoneUtil.kt
package com.example.kargopaylasimkotlin.util

object PhoneUtil {
    fun normalizeTrToE164(input: String): String {
        val raw = input.trim().replace(" ", "").replace("-", "")
        if (raw.isBlank()) return raw

        return when {
            raw.startsWith("+90") -> "+90" + raw.removePrefix("+90").filter { it.isDigit() }
            raw.startsWith("90") && raw.length >= 12 -> "+90" + raw.removePrefix("90").filter { it.isDigit() }
            raw.startsWith("0") -> "+90" + raw.drop(1).filter { it.isDigit() }
            raw.startsWith("5") -> "+90" + raw.filter { it.isDigit() }
            else -> raw // fallback
        }
    }

    fun isLikelyTrPhoneE164(phone: String): Boolean {
        // +90 + 10 hane => 13 char (örn +905xxxxxxxxx)
        if (!phone.startsWith("+90")) return false
        val digits = phone.drop(1).filter { it.isDigit() } // "90xxxxxxxxxx"
        return digits.length == 12
    }
}